lib/src/native/realm_core.dart

```diff
@@ -80,7 +80,12 @@ class _RealmCore {
   }
 
   factory _RealmCore() {
-    return _instance ??= _RealmCore._();
+    _instance ??= _RealmCore._();
+    final logCallback = Pointer.fromFunction<Void Function(Handle, Int32, Pointer<Int8>)>(_logCallback);
+    _realmLib.realm_set_log_callback(
+        logCallback.cast(), Realm.logger.level.toInt(), noopUserdata.toWeakHandle(), _realmLib.addresses.realm_dart_delete_weak_handle);
```

Comment:
We set the logger here in advance in order to have it before the scheduler is initialized.
`realm_dart_delete_weak_handle` is a dummy function, because the userdata_free function is required.